### PR TITLE
Improve SPIRVToLLVM::transOCLBuiltinFromVariable

### PR DESCRIPTION
**Pull request by AlexeySotkin**
The method did not handle correctly the case when addrspacecast instruction had
more than one usage.

Signed-off-by: Alexey Sotkin <alexey.sotkin@intel.com>

### CODE REVIEW

File: lib/SPIRV/SPIRVReader.cpp [hunk 1]
```diff
@@ -265,48 +283,70 @@ bool SPIRVToLLVM::transOCLBuiltinFromVariable(GlobalVariable *GV,
...
+  for (auto *UI : GV->users()) {
+    // There might or might not be an addrspacecast instruction.
+    if (auto *ASCast = dyn_cast<AddrSpaceCastInst>(UI)) {
+      Casts.push_back(ASCast);
+      for (auto *CastUser : ASCast->users() ) {
+        if (auto *LD = dyn_cast<LoadInst>(CastUser)) {
+          FindAndReplace(LD);
+        }
+      }
+    } else if (auto *LD = cast<LoadInst>(UI)) {
```
Comment (svenvh) [inline lib/SPIRV/SPIRVReader.cpp:332]:
```suggestion
    } else if (auto *LD = dyn_cast<LoadInst>(UI)) {
```

Comment (AlexeySotkin) [reply lib/SPIRV/SPIRVReader.cpp]: Thanks! Fixed.

File: lib/SPIRV/SPIRVReader.cpp [hunk 2]
```diff
@@ -265,48 +285,70 @@ bool SPIRVToLLVM::transOCLBuiltinFromVariable(GlobalVariable *GV,
...
+
+  // If HasIndexArg is true, we are going over users of the Load instruction,
+  // which are expected to be ExtractElement instructions. The result of the
+  // ExtractElement instructions is considered as a usage of the GV and should
+  // be replaced with Func.
+  // If HasIndexArg is false, the result of the Load instruction is the value
+  // which should be replaced with the Func.
+  auto FindAndReplace = [&](LoadInst *LD) {
+    Loads.push_back(LD);
+    if (HasIndexArg) {
```
Comment (AlexeySachkov) [inline lib/SPIRV/SPIRVReader.cpp:311]: Minor: suggest to use early exit here:

```c++
if (!HasIndexArg) {
  Replace({}, LD);
  return;
}
for (/* ... */) {
  /* ... */
}
```

Comment (AlexeySotkin) [reply lib/SPIRV/SPIRVReader.cpp:311]: I'd prefer to leave it as is if you don't mind.

Comment (AlexeySachkov) [reply lib/SPIRV/SPIRVReader.cpp:311]: No objections to leave as-is, this is a minor comment

**Pull request merged**